client/http.go

```diff
@@ -0,0 +1,156 @@
+	}
+	c.group = group
+
+	return c, nil
+}
+
+// NewHTTPClientWithGroup constructs an http client when the group parameters are already known.
+func NewHTTPClientWithGroup(url string, group *key.Group, client *http.Client) (Client, error) {
+	c := &httpClient{
+		root:   url,
+		group:  group,
+		client: client,
+	}
+	return c, nil
+}
+
+// httpClient implements Client through http requests to a Drand relay.
+type httpClient struct {
+	root   string
+	client HTTPClient
+	group  *key.Group
+}
+
+// FetchGroupInfo attempts to initialize an httpClient when
+// it does not know the full group paramters for a drand group.
+func (h *httpClient) FetchGroupInfo(groupHash []byte) (*key.Group, error) {
+	// fetch the `Group` to validate connectivity.
+	groupResp, err := h.client.Get(fmt.Sprintf("%s/group", h.root))
+	if err != nil {
+		return nil, err
+	}
+
+	protoGrp := drand.GroupPacket{}
+	if err := json.NewDecoder(groupResp.Body).Decode(&protoGrp); err != nil {
+		return nil, err
+	}
+	grp, err := key.GroupFromProto(&protoGrp)
+	if err != nil {
+		return nil, err
+	}
+
+	if grp.PublicKey == nil {
+		return nil, fmt.Errorf("Group does not have a valid key for validation")
+	}
+
+	if groupHash != nil && !bytes.Equal(grp.Hash(), groupHash) {
+		return nil, fmt.Errorf("%s does not advertise the expected drand group", h.root)
+	}
+	return grp, nil
+}
+
+// Implement textMarshaller
+func (h *httpClient) MarshalText() ([]byte, error) {
+	return json.Marshal(h)
+}
+
+type randData struct {
+	Round             uint64 `json:"round,omitempty"`
+	Signature         []byte `json:"signature,omitempty"`
+	PreviousSignature []byte `json:"previous_signature,omitempty"`
+	Randomness        []byte `json:"randomness,omitempty"`
+}
+
+// Get returns a the randomness at `round` or an error.
+func (h *httpClient) Get(ctx context.Context, round uint64) (Result, error) {
+	randResponse, err := h.client.Get(fmt.Sprintf("%s/public/%d", h.root, round))
+	if err != nil {
+		return Result{}, err
+	}
+
+	randResp := randData{}
+	if err := json.NewDecoder(randResponse.Body).Decode(&randResp); err != nil {
+		return Result{}, err
+	}
+	if len(randResp.Signature) == 0 || len(randResp.PreviousSignature) == 0 {
+		return Result{}, fmt.Errorf("insufficent response")
+	}
+
+	b := beacon.Beacon{
+		PreviousSig: randResp.PreviousSignature,
+		Round:       randResp.Round,
+		Signature:   randResp.Signature,
+	}
+	if err := beacon.VerifyBeacon(h.group.PublicKey.Key(), &b); err != nil {
+		return Result{}, err
+	}
+
+	return Result{Round: randResp.Round, Signature: randResp.Signature}, nil
+}
+
+// Watch returns new randomness as it becomes available.
+func (h *httpClient) Watch(ctx context.Context) <-chan Result {
+	ch := make(chan Result, 1)
+	r := h.RoundAt(time.Now())
+	val, err := h.Get(ctx, r)
+	if err != nil {
+		ch <- val
+	} else {
+		close(ch)
+		return ch
+	}
+	time.AfterFunc(time.Unix(beacon.TimeOfRound(h.group.Period, h.group.GenesisTime, r+1), 0).Sub(time.Now()), func() {
```

Comment:
You can use [`NextRound`](https://github.com/drand/drand/blob/master/beacon/beacon.go#L112) to compute the time of the next round to make:
```
_,nextTime := beacon.NextRound(time.Now().Unix(),h.group.GenesisTime, h.group.Period)
time.Sleep(time.Duration(nextTime - time.Now().Unix()) * time.Second)
```
I find it a bit clearer... ?

Reply:
The disjoint between this interface (seconds since unix epoch) and go's `time.Time` is leading to a lot of conversions.

Would it be worth having beacon interfaces for directlying accessing times/durations in native golang time formats?